hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/DroolsRulesEngineImpl.java

```diff
@@ -155,20 +157,18 @@ public void fire() {
         // execution of the rules.  So, if we find multiple Data instances for the same Id, defer all but
         // the oldest to a subsequent run. Note that pendingData is already sorted by (id ASC, timestamp ASC) so
         // the iterator will present Data with the same id together, and time-ordered.
-        log.debug("firing rules...");
+        int initialPendingData = pendingData.size();
+        int initialPendingEvents = pendingEvents.size();
+        log.debugf("Firing rules... PendingData [%s] PendingEvents [%s]", initialPendingData, initialPendingEvents);
```

Comment:
initialPending* variables are used at the end of the computation to show final message.